src/sql/workbench/parts/dataExplorer/common/nodeContextUtils.ts

```diff
@@ -0,0 +1,156 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import { INodeContextValue } from 'sql/workbench/parts/dataExplorer/common/nodeContext';
+import { RawContextKey, IContextKeyService, IContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
+import { mssqlProviderName } from 'sql/platform/connection/common/constants';
+import { NodeType } from 'sql/workbench/parts/objectExplorer/common/nodeType';
+
+export class NodeContextUtils extends Disposable {
+
+	static readonly canSelect = new Set([NodeType.Table, NodeType.View]);
+	static readonly canEditData = new Set([NodeType.Table]);
+	static readonly canCreateOrDelete = new Set([NodeType.AggregateFunction, NodeType.PartitionFunction, NodeType.ScalarValuedFunction,
+	NodeType.Schema, NodeType.StoredProcedure, NodeType.Table, NodeType.TableValuedFunction,
+	NodeType.User, NodeType.UserDefinedTableType, NodeType.View]);
+	static readonly canExecute = new Set([NodeType.StoredProcedure]);
+	static readonly canAlter = new Set([NodeType.AggregateFunction, NodeType.PartitionFunction, NodeType.ScalarValuedFunction,
+	NodeType.StoredProcedure, NodeType.TableValuedFunction, NodeType.View]);
```

Comment:
This should really be something that is defined by the nodes themselves - tying it here just makes it harder to keep up to date and creates unneeded dependencies.

Reply:
Even in our main OE service, we have a mapping. We don't have a way for the nodes to tell their scripting actions right now.